packages/fxa-payments-server/server/lib/routes/legal-docs.js

```diff
@@ -0,0 +1,184 @@
+    const entry = { checkedOn: Date.now(), result: res.status === 200 };
+    cache.set(url, entry);
+    return entry['result'];
+  } catch {
+    // NOOP
+    // fetch API exceptions are not 3xx-5xx responses
+  }
+
+  return false;
+};
+
+module.exports = {
+  method: 'get',
+  path: '/legal-docs',
+  validate: {
+    query: { url: joi.string().uri().required() },
+  },
+  async process(req, res) {
+    const docUrl = new URL(req.query.url);
+
+    // Limit the redirect to just the CDN that's hosting the legal docs
+    if (docUrl.hostname !== cdnFqdn) {
+      return res.sendStatus(400).end();
+    }
+
+    // For backwards compatibility and future single-locale products(?)
+    if (docUrl.pathname.endsWith('.pdf')) {
+      return res.redirect(docUrl.href).end();
+    }
+
+    const reqLanguage = req.get('accept-language')
+      ? req.get('accept-language')
+      : DEFAULT_LOCALE;
+    const parsed = acceptLanuageParser.parse(reqLanguage);
+
+    if (!parsed.length) {
+      return res.sendStatus(400).end();
+    }
+
+    const withTag = parsed.map((l) => ({ ...l, tag: getLanguageTag(l) }));
+
+    // Since a request here potentially fans out to mulitple requests, limit that number
+    const supported = withTag
+      .filter((l) => isSupportedLanguage(l.tag))
+      .slice(0, LANGUAGES_LIMIT);
+
+    // Check for exact match on the language tag in order of priority
+    const checkedTags = [];
+    for (const l of supported) {
+      const downloadUrl = new URL(
+        getFullPath(docUrl.pathname, l.tag),
+        docUrl.origin
+      );
+
+      if (await isValidDocUrl(downloadUrl.href)) {
+        downloadUrl.search = docUrl.search;
+        return res.redirect(downloadUrl.href).end();
+      }
+
+      checkedTags.push(l.tag);
+    }
+
+    // Fallback: for those with a region, try the language without the region
+    const languageOnlyTags = supported.filter(
+      (l) => !!l.region && !checkedTags.includes(l.code)
+    );
+
+    for (const l of languageOnlyTags) {
+      const downloadUrl = new URL(
+        getFullPath(docUrl.pathname, l.code),
+        docUrl.origin
+      );
+
+      if (await isValidDocUrl(downloadUrl.href)) {
+        downloadUrl.search = docUrl.search;
+        return res.redirect(downloadUrl.href).end();
+      }
+    }
+
+    // Fallback: try default locale
+    const defaultDocUrl = new URL(
+      getFullPath(docUrl.pathname, DEFAULT_LOCALE),
+      docUrl.origin
+    );
+    if (await isValidDocUrl(defaultDocUrl.href)) {
+      defaultDocUrl.search = docUrl.search;
+      return res.redirect(defaultDocUrl.href).end();
+    }
+
+    // :(
+    Sentry.withScope((scope) => {
+      scope.setContext('subscription.legalDocs', {
+        url: docUrl.href,
+        acceptLanguage: req.get('accept-language'),
+      });
+      Sentry.captureMessage(
+        `Legal doc redirect failed.`,
+        Sentry.Severity.Warning
+      );
+    });
+    return res.sendStatus(404).end();
```

Comment:
Does it maybe make sense, if all else fails, to redirect to mozilla.org/privacy or mozilla.org/legal?

Reply:
I don't know.  Feels like a product and/or legal question.  Good idea though.

@clouserw What do  you think?

Reply:
Our fallback should be English.  Beyond that, (eg. if English doesn't exist) I don't think we need more redirects, but it would be nice to alert Sentry so we could see the error.

Reply:
Yeah, the Sentry reporting code is just above the line in question.  It captures the requested url (from the metadata when it's legit) and the value of accept-language.